volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -374,6 +374,35 @@ def get_threads(self) -> Iterable[interfaces.objects.ObjectInterface]:
         ):
             yield task
 
+    def get_pid(self) -> int:
+        """Returns the pid of this process"""
+        return self.tgid
+
+    def get_parent_pid(self) -> int:
+        """Returns the pid of parent of this process"""
+        # Uses real_parent rather than parent to match Linux kernel getppid
+        # /*
+        #  * Accessing ->real_parent is not SMP-safe, it could
+        #  * change from under us. However, we can use a stale
+        #  * value of ->real_parent under rcu_read_lock(), see
+        #  * release_task()->call_rcu(delayed_put_task_struct).
+        #  */
+        # SYSCALL_DEFINE0(getppid)
+        # {
+        # 	int pid;
+
+        # 	rcu_read_lock();
+        # 	pid = task_tgid_vnr(rcu_dereference(current->real_parent));
+        # 	rcu_read_unlock();
+
+        # 	return pid;
+        # }
+        return self.real_parent.get_pid() if self.parent else 0
```

Comment:
`self.real_parent` in both. Additionally, `parent` and `real_parent` are pointers. So, it could be useful to also check the pointer is readable.
```python
return self.real_parent.get_pid() if self.real_parent and self.real_parent.is_readable() else 0
```

Lastly, there is no need to paste the entire code in the comments. Simply referencing the kernel function name will suffice. Including it could cause confusion, as someone might assume we need synchronization primitives like locks or mutexes, even though thread-safety isn't an issue for us.